### PR TITLE
Add System Hook

### PR DESCRIPTION
**Pull request by golflima**
Hi,

This PR add support for the Gitlab System Hooks (if you use Gitlab CE or EE).
We use it at Virtua SA to monitor our Gitlab activity.
The rocketchat messages can be improved, let me know if you want us to make changes.

Best regards,
Jérémy WALTHER.

### CODE REVIEW

**Comment by golflima** [comment]
I'll update the PR later ...
We are upgrading our gitlab, and it seems there is now more events to handle in System hook: https://docs.gitlab.com/ce/system_hooks/system_hooks.html
I should be able to add them before the end of the week.

File: gitlab-rocketchat.hooks.js [hunk 1]
```diff
@@ -377,7 +377,8 @@ See: ${data.object_attributes.url}`
 		switch (event_name) {
 			case 'project_create':
 			case 'project_destroy':
-				action = event_name === 'project_create' ? 'created' : 'removed';
+			case 'project_update':
+				action = event_name === 'project_create' ? 'created' : (event_name === 'project_create' ? 'removed' : 'updated');
```
Comment (malko) [inline gitlab-rocketchat.hooks.js]: I think it should be (event_name === 'project_destroy') here

File: gitlab-rocketchat.hooks.js [hunk 2]
```diff
@@ -362,4 +365,63 @@ See: ${data.object_attributes.url}`
 			}
 		};
 	}
+
+	systemEvent(data) {
+		const created_at = data.created_at;
```
Comment (malko) [inline gitlab-rocketchat.hooks.js]: using destructuring would have been nice

**Review by malko — COMMENTED** [review]
Not sure this really reduce complexity for the human reading the code ;)

**Comment by malko** [comment]
Thanks for your contrib, I'll review it more deeply really soon, i'm not sure your last commit is really necessary.

File: gitlab-rocketchat.hooks.js (hunk 1 above)
Comment (golflima) [reply gitlab-rocketchat.hooks.js]: Indeed, was fixed in the last commit.

**Comment by golflima** [comment]
You're welcome. Thanks you a lot for creating this script in first place, it is very useful.

Yep, last commit was just for complying to your eslint, but is not really less complex...

There are two more things to do later (in another PR?):

* Implement `push`, `tag` and `repository_update` sub system events ... but we won't use them and I don't know yet how to render them in a *friendly* way with rocketchat ...
* Filter unnecessary keys of `data` before attaching it to the rochatchat messages (such as `created_at`) or auto-collapse attachments but it is still an open request on rocketchat

**Comment by malko** [comment]
Thank you again, and sorry for the delay.
Other PR are welcome

**Comment by golflima** [comment]
Nice refactoring.
I'll send you another PR soon, because we discovered other non-documented Gitlab system events ...

**Pull request merged**